src/widgets/modalChildren/BreachDisplay.js

```diff
@@ -0,0 +1,97 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React from 'react';
+import { connect } from 'react-redux';
+import { View, FlatList, StyleSheet, ActivityIndicator } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { getPageInfoColumns } from '../../pages/dataTableUtilities';
+import { chunk } from '../../utilities/chunk';
+
+import { BreachChart } from '../BreachChart';
+import { PageInfo } from '../PageInfo';
+
+import { WHITE, SUSSOL_ORANGE } from '../../globalStyles';
+
+const Breach = ({ breach, getInfoColumns }) => {
+  const { temperatureLogs } = breach;
+
+  const [chartData, setChartData] = React.useState({ lineData: [], render: false });
+  const { render, lineData } = chartData;
+
+  React.useEffect(() => {
+    const { length: numberOfLogs } = temperatureLogs;
+    setTimeout(() => {
+      const chunkSize = Math.ceil(numberOfLogs / Math.min(numberOfLogs, 5));
+      const chunkedLogs = chunk(temperatureLogs, chunkSize);
+      const mappedLineData = chunkedLogs.map(chunkOfLogs => ({
+        temperature: Math.max(...chunkOfLogs.map(({ temperature }) => temperature)),
+        timestamp: Math.max(...chunkOfLogs.map(({ timestamp }) => timestamp)),
+      }));
+
+      setChartData({ lineData: mappedLineData, render: true });
+    }, 1000);
```

Comment:
bit hacky - displays a spinner for a second and then switches to the chart 🤷‍♂

Reply:
yeah, maybe not ideal. but works.